coffee/base.py

```diff
@@ -435,24 +435,27 @@ class Symbol(Expr):
     * 1: array
     * 2: matrix, etc.
 
-    :param tuple rank: entries represent the iteration variables the symbol
+    :param symbol: the Symbol name.
+    :param rank: entries represent the iteration variables the symbol
         depends on, or explicit numbers representing the entry of a tensor the
         symbol is accessing, or the size of the tensor itself.
-    :param list offset: an iterator of 2-tuple (period, stride), one 2-tuple for
+    :param offset: an iterator of 2-tuple (period, stride), one 2-tuple for
         each entry in rank. The period is the multiplier of the rank, while
         stride is a quantity summed to the rank. E.g., if rank=(i, j) and
         offset=((1, 0), (3, 2)), printing a symbol 'a' returns the string
         a[i][3*j + 2].
+    :param operator: an optional C operator to be applied (e.g., *, &).
     """
 
-    def __init__(self, symbol, rank=(), offset=()):
+    def __init__(self, symbol, rank=(), offset=(), operator=None):
         super(Symbol, self).__init__([])
         self.symbol = symbol
         self.rank = rank
         self.offset = offset or tuple([(1, 0) for r in rank])
+        self.operator = operator
 
     def operands(self):
-        return [self.symbol, self.rank, self.offset], {}
+        return [self.symbol, self.rank, self.offset, self.operator], {}
```

Comment:
As an aside note: I'm not saying and I don't think that refactoring that part of the hierarchy would be difficult (operators, shape, etc) , just that it would take a non-negligible (for me, in this period) amount of time